### PR TITLE
Add support for preemptible instances.

### PR DESCRIPTION
**Pull request by jadair-cloudera**
Added configuration parameter, allocation logic, and tests.

### CODE REVIEW

File: tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import static com.cloudera.director.google.compute.GoogleComputeInstanceTemplateConfigurationProperty.NETWORK_NAME;
 import static com.cloudera.director.google.compute.GoogleComputeInstanceTemplateConfigurationProperty.TYPE;
 import static com.cloudera.director.google.compute.GoogleComputeInstanceTemplateConfigurationProperty.ZONE;
+import static com.cloudera.director.google.compute.GoogleComputeInstanceTemplateConfigurationProperty.USE_PREEMPTIBLE_INSTANCES;
```
Comment (duftler) [inline tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java]: Sort-order is off here.

File: tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java [hunk 2]
```diff
@@ -106,6 +107,15 @@ public GoogleComputeProviderFullCycleTest(String localSSDInterfaceType, String i
...
   public void testFullCycle() throws InterruptedException, IOException {
+    doTestFullCycle(null);
+  }
+
+  @Test
+  public void testFullCyclePreemptible() throws InterruptedException, IOException {
+    doTestFullCycle(true);
+  }
+
+  public void doTestFullCycle(Boolean usePreemptibleInstances) throws InterruptedException, IOException {
```
Comment (duftler) [inline tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java]: What's the reason behind using the wrapper `Boolean` instance of `boolean` and dropping the null-check?

File: tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java (hunk 1 above)
Comment (jadair-cloudera) [inline tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java]: Edited the wrong one after copy-paste. :( Will fix.

File: tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java (hunk 2 above)
Comment (jadair-cloudera) [inline tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java]: To verify that the defaulting behavior works correctly.

Comment (duftler) [inline tests/src/test/java/com/cloudera/director/google/compute/GoogleComputeProviderFullCycleTest.java]: Ah, makes sense.

**Comment by duftler** [comment]
Looks good:
image

**Comment by duftler** [comment]
LGTM.

**Pull request merged**